src/main/java/Duke.java

```diff
@@ -1,10 +1,239 @@
+        }
+        String command = input.substring(6);
+        if(command.contains (" /at ")){
+            String[] parts = command.split(" /at ");
+            String description = parts[0];
+            String date = parts[1];
+            Event t = new Event(description,date);
+            list[LIST_COUNTER] = t;
+            LIST_COUNTER += 1;
+            System.out.println("I have added [" + t.getType() + "]["
+                + t.getStatusIcon() + "] \""
+                + t.getName() + t.getDate() + "\"" + " to the List!");
+            printNoOfTask();
+        } else {
+            error(5);
+        }
+    }
+
+    /**
+     * Check if done command is valid
+     * @param input - index of task
+     */
+    private static void doneTask(String input) {
+        if (input.equalsIgnoreCase("done")) {
+            error(1);
+            return;
+        }
+        if (input.substring(5).matches("[0-9]+")) {
+            int index = Integer.parseInt(input.substring(5));
+            if (index > LIST_COUNTER) {
+                error(2);
+                return;
+            }
+            checkTask(index - 1);
+        } else if (input.substring(5).isBlank()) {
+            error(1);
+        } else {
+            error(1);
+        }
+    }
+
+    /**
+     * Check respective task as done
+     * @param index - index of list
+     */
+    public static void checkTask(int index){
+        if (list[index].isDone) {
+            error(4);
+        } else {
+            System.out.println("Good Job, I will mark this as done!");
+            list[index].markAsDone();
+            System.out.println("[" + list[index].getType() + "] ["
+                + list[index].getStatusIcon() + "] " + list[index].getName()
+                + list[index].getDate());
+            PrintOutput.printBorder();
+        }
+    }
+
+    /**
+     * Print the number of task in list
+     */
+    public static void printNoOfTask() {
+        System.out.print("You have " + LIST_COUNTER + " task");
+        if(LIST_COUNTER > 1){
+            System.out.print("s");
+        }
+        System.out.print(" in total!\n");
+        PrintOutput.printBorder();
+    }
+
+    /**
+     * print List
+     */
+    public static void printList(){
+        if (LIST_COUNTER > 0) {
+            System.out.println(" LIST");
+            for (int i = 0; i < LIST_COUNTER; i++) {
+                System.out.println(i + 1 +  ". [" + list[i].getType() +  "] " + "["
+                    +list[i].getStatusIcon() + "] " + list[i].getName()
+                    + list[i].getDate());
+            }
+            PrintOutput.printBorder();
+        } else {
+            error(3);
+        }
+    }
+
+
+    /**
+     * Returns error code message
+     * Error Code 1: done command in wrong format
+     * Error Code 2: done Index is more than size of array
+     * Error Code 3: Timetable is empty
+     * Error Code 4: Task already marked done
+     * Error Code 5: Invalid Command format
```

Comment:
I would suggest use constants or enum to represent these error codes (as they are magic numbers).
It is hard to read from code if something like `error(5)` is written.